sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -854,18 +854,26 @@ protected virtual TimeSpan CalculateLinkAuthorizationRefreshInterval(DateTime ex
         ///   credentials.
         /// </remarks>
         ///
-        protected virtual Task<DateTime> RequestAuthorizationUsingCbsAsync(AmqpConnection connection,
-                                                                           CbsTokenProvider tokenProvider,
-                                                                           Uri endpoint,
-                                                                           string audience,
-                                                                           string resource,
-                                                                           string[] requiredClaims,
-                                                                           TimeSpan timeout)
+        protected virtual async Task<DateTime> RequestAuthorizationUsingCbsAsync(AmqpConnection connection,
+                                                                                 CbsTokenProvider tokenProvider,
+                                                                                 Uri endpoint,
+                                                                                 string audience,
+                                                                                 string resource,
+                                                                                 string[] requiredClaims,
+                                                                                 TimeSpan timeout)
         {
             try
             {
+                // If the connection is in the process of closing, then do not attempt to authorize but consider it an
+                // unexpected scenario that should be treated as transient.
+
+                if (connection.IsClosing())
+                {
+                     throw new EventHubsException(true, EventHubName, Resources.UnknownCommunicationException, EventHubsException.FailureReason.ServiceCommunicationProblem);
```

Comment:
In this case is it appropriate to call this an `UnknownCommunicationException`, or is a closing connection not a deterministic condition?

Reply:
It's non-deterministic;  the service can force-close the connection in unpredictable ways.  For example, they'll close every 20(ish) minutes to ensure clients are active or when nodes are migrated because the host machines are being patched.   

If we've made it to this point, the client thinks that the connection should be open and healthy; this check is intended only to avoid the overhead of triggering an exception if we can prove our assumption about connection health is incorrect.

Reply:
That said, if you have a suggestion for a better reason, I'm open to it.   This is basically the most generic that we have, which I chose just because we don't know exactly why the connection is closing.